ui/server/src/main/scala/pl/touk/nussknacker/ui/db/entity/ProcessDeploymentInfoEntityFactory.scala

```diff
@@ -8,77 +8,76 @@ import pl.touk.nussknacker.restmodel.processdetails.DeploymentAction.DeploymentA
 
     def pk = primaryKey("pk_deployed_process_version", (processId, processVersionId, environment, deployedAt))
 
-    def processes_fk = foreignKey("proc_ver_in_deployed_proc_fk", (processId, processVersionId), processVersionsTable)(
+    def processes_fk: ForeignKeyQuery[ProcessVersionEntityFactory#ProcessVersionEntity, ProcessVersionEntityData] = foreignKey("proc_ver_in_deployed_proc_fk", (processId, processVersionId), processVersionsTable)(
       procV => (procV.processId, procV.id.?),
       onUpdate = ForeignKeyAction.Cascade,
       onDelete = ForeignKeyAction.NoAction
     )
 
-    def comment_fk = foreignKey("comment_in_deployed_proc_fk", commentId, commentsTable)(
+    def comment_fk: ForeignKeyQuery[CommentEntityFactory#CommentEntity, CommentEntityData] = foreignKey("comment_in_deployed_proc_fk", commentId, commentsTable)(
       _.id.?,
       onUpdate = ForeignKeyAction.Cascade,
       onDelete = ForeignKeyAction.SetNull
     )
 
-    def environment_fk = foreignKey("env_in_deployed_proc_fk", environment, environmentsTable)(
+    def environment_fk: ForeignKeyQuery[EnvironmentsEntityFactory#EnvironmentsEntity, EnvironmentsEntityData] = foreignKey("env_in_deployed_proc_fk", environment, environmentsTable)(
       _.name,
       onUpdate = ForeignKeyAction.Cascade,
       onDelete = ForeignKeyAction.NoAction
     )
 
-    def * = (processId, processVersionId, environment, user, deployedAt, deploymentAction, commentId, buildInfo) <> (
-      DeployedProcessVersionEntityData.tupled, DeployedProcessVersionEntityData.unapply)
-
+    def * : ProvenShape[DeployedProcessInfoEntityData] = (processId, processVersionId, environment, user, deployedAt, deploymentAction, commentId, buildInfo) <> (
+      DeployedProcessInfoEntityData.tupled, DeployedProcessInfoEntityData.unapply)
   }
-
-  val deployedProcessesTable: LTableQuery[ProcessDeploymentInfoEntityFactory#ProcessDeploymentInfoEntity] = 
-    LTableQuery(new ProcessDeploymentInfoEntity(_))
-
-  implicit def deploymentMapper: JdbcType[DeploymentAction] with BaseTypedType[DeploymentAction] = MappedColumnType.base[DeploymentAction, String](
-    _.toString,
-    DeploymentAction.withName
-  )
-
 }
 
-case class DeployedProcessVersionEntityData(processId: Long,
-                                            processVersionId: Option[Long],
-                                            environment: String,
-                                            user: String,
-                                            deployedAt: Timestamp,
-                                            deploymentAction: DeploymentAction,
-                                            commentId: Option[Long],
-                                            buildInfo: Option[String]) {
-  val deployedAtTime: LocalDateTime = DateUtils.toLocalDateTime(deployedAt)
+case class DeployedProcessInfoEntityData(processId: Long,
+                                         processVersionId: Option[Long],
```

Comment:
I know that it already was this way, but why processVersionId was optional?

Reply:
Good question. I will check it ;)

Reply:
I'v changed it. Now it's Long and I'v added migrations for hsql and postgresql.